src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/IllegalImportCheck.java

```diff
@@ -118,16 +165,39 @@ public void visitToken(DetailAST ast) {
     }
 
     /**
-     * Checks if an import is from a package that must not be used.
+     * Checks if an import is from a package or class name that must not be used.
      * @param importText the argument of the import keyword
      * @return if {@code importText} contains an illegal package prefix
      */
     private boolean isIllegalImport(String importText) {
-        for (String element : illegalPkgs) {
-            if (importText.startsWith(element + ".")) {
-                return true;
+        boolean result = false;
+        if (regexp) {
+            for (Pattern pattern : illegalPkgsRegexps) {
+                if (pattern.matcher(importText).matches()) {
```

Comment:
Should an illegal package search scan the entire import text, which has the class too?